src/targets/gpu/device/multinomial.cpp

```diff
@@ -0,0 +1,54 @@
+#include <migraphx/shape.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/dfor.hpp>
+#include <migraphx/gpu/device/multinomial.hpp>
+#include <migraphx/gpu/device/tensor.hpp>
+#include <migraphx/gpu/device/launch.hpp>
+#include <migraphx/gpu/device/types.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+namespace device {
+
+template <class T1, class T2>
+void __device__
+set_output(T1& cdf, T1& dist, T2& output, const size_t& i, size_t& j, const size_t& n)
+{
+    for(; j < n; ++j)
+    {
+        if(cdf[j] > (dist[i] * cdf[n - 1]))
+        {
+            output[i] = j;
```

Comment:
Compared to the ref implementation, I think this line should be

```
output[i] = j - start_of_j;
```
But why the test passed?

Reply:
You're right. I think the test was passing because it was for a single batch, which means `start_of_j` was just 0.

Reply:
Fixed and updated the test